src/bitmap/bitmap_ops.rs

```diff
@@ -85,18 +83,20 @@ where
     assert_eq!(lhs.len(), rhs.len());
     let mut lhs_chunks = lhs.chunks();
     let mut rhs_chunks = rhs.chunks();
+    let rem_lhs = lhs_chunks.remainder();
+    let rem_rhs = rhs_chunks.remainder();
 
     let chunks = lhs_chunks
         .by_ref()
         .zip(rhs_chunks.by_ref())
         .map(|(left, right)| op(left, right));
-    // Soundness: `BitChunks` is a trusted len iterator
-    let mut buffer = unsafe { MutableBuffer::from_chunk_iter_unchecked(chunks) };
 
-    let remainder_bytes = lhs_chunks.remainder_len().saturating_add(7) / 8;
-    let rem = op(lhs_chunks.remainder(), rhs_chunks.remainder());
-    let rem = &rem.to_ne_bytes()[..remainder_bytes];
-    buffer.extend_from_slice(rem);
+    // Soundness: `BitChunks` is a trusted len iterator
+    let buffer = unsafe {
+        MutableBuffer::from_chunk_iter_unchecked(
+            chunks.chain(std::iter::once(op(rem_lhs, rem_rhs))),
+        )
+    };
```

Comment:
```suggestion
    let lhs_chunks = lhs.chunks();
    let rhs_chunks = rhs.chunks();

    let rem = op(lhs_chunks.remainder(), rhs_chunks.remainder());

    let chunks = lhs_chunks
        .zip(rhs_chunks)
        .map(|(left, right)| op(left, right));
    let chunks = chunks.chain(std::iter::once(rem));
    let buffer = MutableBuffer::from_chunk_iter(chunks);
```

so that we do not need the `unsafe`: zip and `chain(once(` is trusted len and we can use the traits to get that one if we do not use `by_ref`

Reply:
Thanks - great suggestion!